### PR TITLE
veahavta home assignment- Ziv Bryk

### PR DESCRIPTION
**Pull request by zivbryk**
Hi, 
> Desktop version is completed (form validations, msg on form submit, scroll to section on link clicks, email opens on 
   address button)
> Mobile version: header + hamburger menu completed, still need to convert all other section to 2 cols grid layout

Happy Passover!

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202137838393519

### CODE REVIEW

File: components/data-components/contact-form.tsx [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+    email: '',
+    phone: '',
+    message: '',
+  })
+
+  const [isInputValid, setIsInputValid] = useState(false)
+  const [isSubmitMsg, setIsSubmitMsg] = useState(false)
+
+  {
+    /* Dynamically render a component if time permits */
```
Comment (omrirz) [inline components/data-components/contact-form.tsx]: Remove the commented code we don't need it

File: components/data-components/contact-form.tsx [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+                  className="text-red pt-2 text-xl rtl:pr-2 ltr:pl-2"
+                />
+              </div>
+
+              <div className="flex flex-col">
+                <label htmlFor="phone" className="text-2xl font-bold pb-3">
+                  {data.common.contactUsFormPhone}
+                </label>
+                <Field
+                  type="phone"
```
Comment (omrirz) [inline components/data-components/contact-form.tsx]: should be type tel

File: components/data-components/contact-form.tsx [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+                  className="text-red pt-2 text-xl rtl:pr-2 ltr:pl-2"
+                />
+              </div>
+
+              <div className="flex flex-col">
+                <label htmlFor="lastname" className="text-2xl font-bold pb-3">
+                  {data.common.contactUsFormLastName}
+                </label>
+                <Field
+                  type="lastname"
```
Comment (omrirz) [inline components/data-components/contact-form.tsx]: type text

File: components/data-components/contact-form.tsx [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+                  />
+                )
+              })} */}
+
+              <div className="flex flex-col">
+                <label htmlFor="firstname" className="text-2xl font-bold pb-3">
+                  {data.common.contactUsFormFirstName}
+                </label>
+                <Field
+                  type="firstname"
```
Comment (omrirz) [inline components/data-components/contact-form.tsx]: type text

File: components/data-components/contact-us-link.tsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+      return link.linkValue.replace(/\+/g, '') + '+'
+    } else {
+      return link.linkValue
+    }
+  }
+
+  const onLinkClick = () => {
+    if (link.linkType === 'email') {
+      //For Gmail (user has to be signed in to google)
+      window.location.href =
```
Comment (omrirz) [inline components/data-components/contact-us-link.tsx]: this is not they way to do that.
Here is how an email link should be:

```
<a href="mailto:email@domain.com" className =grid grid-cols-auto-1fr gap-x-2">
    <Icon />
    <Text />
</a>
```


Also other links should be:

Phone:
```
<a href="tel:+123456789" className =grid grid-cols-auto-1fr gap-x-2">
    <Icon />
    <Text />
</a>
```

Address:
```
<a href="mailto:href="https://www.google.co.il/maps/search/מגדל הנביאים, ח׳ורי 2, קומה 6, חיפה"" className =grid grid-cols-auto-1fr gap-x-2">
    <Icon />
    <Text />
</a>
```

File: components/footer.tsx [hunk 6]
```diff
@@ -1,12 +1,94 @@
...
+            <div className="footer-menu flex flex-col items-start justify-start">
+              <h2 className="text-xl font-bold mb-2">{data.footerMenuTitle}</h2>
+              <ul>
+                <li></li>
+                <li>{data.footerContactUsTitle}</li>
+                <li>{data.footerSupportUsButton}</li>
+              </ul>
+            </div>
+
+            <div className="footer-contact flex flex-col items-center">
```
Comment (omrirz) [inline components/footer.tsx]: why do we need "footer-contact" class here?

File: components/header.tsx [hunk 7]
```diff
@@ -1,43 +1,171 @@
...
+        }
+        span:nth-child(2),
+        span:nth-child(3) {
+          top: 10px;
+        }
+        span:nth-child(4) {
+          top: 20px;
+        }
+
+        // div.nav-icon.open {
```
Comment (omrirz) [inline components/header.tsx]: remove commented code. do it everywhere

File: components/footer.tsx [hunk 8]
```diff
@@ -1,12 +1,94 @@
...
+            </div>
+          </div>
+        </footer>
+      </WrapperLarge>
+
+      <style jsx>{`
+        div.logo {
+          background-image: url('assets/img/logo.svg');
+        }
+        span.footer-contact-img {
```
Comment (omrirz) [inline components/footer.tsx:91]: why do we need this?

File: components/header.tsx [hunk 9]
```diff
@@ -1,43 +1,171 @@
...
-            </li>
-            <li>
-              <ChangeLangButton className="" lang="he">
-                {data.languageNames[0].he}
-              </ChangeLangButton>
-            </li>
-          </ul>
+      <header className="absolute w-full z-10 sm:pt-3 sm:px-3 pt-1 px-1">
+        <div className="sm:grid sm:grid-cols-auto-1fr sm:gap-x-6 sm:mx-auto sm:max-w-screen-lg sm:py-1 sm:px-3 bg-light rounded-lg h-22 sm:items-center sm:relative">
+          <div className="flex justify-between p-1 sm:p-0">
```
Comment (omrirz) [inline components/header.tsx:47]: hamburger menu looks like a good candidate for a separate component. why clutter the header with so much html css and functionality?

File: lib/interface.tsx [hunk 10]
```diff
@@ -1,14 +1,31 @@
+import { array } from 'yup'
+
 export type HomePageType = {
   data: {
     homepage: Homepage
     common: Common
   }
 }
+export type WhatWeDoCardType = {
+  card: {
```
Comment (omrirz) [inline lib/interface.tsx:10]: This is nice but if you did it here you should remove it from line 40 and use that type there instead. now you got it duplicated.

File: pages/index.tsx [hunk 11]
```diff
@@ -23,8 +23,8 @@ export default function Home({ data }: HomePageType) {
       <AboutTheClinic data={data} />
       <WhatWeDo data={data} />
       <ContactUs data={data} />
-      Here is the structure of the home page data:
-      <pre className="bg-dark text-light">{JSON.stringify(data, null, 2)}</pre>
+      {/* Here is the structure of the home page data:
```
Comment (omrirz) [inline pages/index.tsx]: delete things you don't need. we can find it later if needed using git.

**Review by omrirz — COMMENTED** [review]
Overall it looks nice. Good job.
There are a few things we need to address.

1.The navbar has an issue in english
Screen Shot 2022-04-21 at 15 01 00

2. align the links vertically to center and why only the icon is clickable? what about the text? (also this is not the way to make links, see the comment in the code)
Screen Shot 2022-04-21 at 14 59 42

3. I submitted the form empty and then I started typing my last name. Why do I still have a validation error, It should bee removed once the user start typing and not on blur?
Screen Shot 2022-04-21 at 15 27 56

4. Note that all the links should be clickable (the icon and text together).
Screen Shot 2022-04-21 at 15 01 39
Screen Shot 2022-04-21 at 15 01 41

5. when I click contact us it scrolls down to the component but part of the form is not on screen
Screen Shot 2022-04-21 at 15 30 15

6. The smooth scroll doesn't work on mobile. Maybe we should use something like https://github.com/iamdustan/smoothscroll

7. links should be links (not buttons and texts)
Screen Shot 2022-04-21 at 15 32 26

File: components/data-components/contact-form.tsx (hunk 1 above)
Comment (zivbryk) [reply components/data-components/contact-form.tsx]: done

File: components/data-components/contact-form.tsx (hunk 2 above)
Comment (zivbryk) [reply components/data-components/contact-form.tsx]: done

File: components/header.tsx (hunk 9 above)
Comment (zivbryk) [reply components/header.tsx:47]: done

**Comment by zivbryk** [comment]
Hi Omri,
Thanks for the elaborate review!
1. navbar was working fine until I made changes for mobile, will find the
bug and fix it
2. "Contact us" links - I have some bug with the vertical alignment - upon
first rendering it is aligned but when I refresh it gets out of order -
will look into it and also will make the whole text clickable. Regarding
the link itself - I left there a commented "mailto:" link as an option
because it opens the user's default email and for me it's outlook which I
don't use and even after I made gmail my default it still opens outlook :).
when I use the other approach  it works and opens my email. one
disadvantage is that if the user is not signed in to his account the link
won't  work. In addition - why did you use "mailto:" also for the address
link? I pasted it in the url/address bar and it didn't work...
3. Will fix validations
4. "all the links should be clickable" - do you also mean phone and
address? open map on address click? What to do on phone click?
5. Easy fix - i'll move the id attribute to the top of the form
6. I'll implement a smooth scroll on mobile
7. "links should be links" - you mean convert the <li> to link under "menu"
in the footer? cause in your snip there were also the the two buttons with
the "לקביעת תור" & "תמכו בנו" - you didn't refer to those purple buttons
right?
** I'll clean the rest of the code according to your comments.
Cheers,
Ziv

**Pull request closed without merging**